packages/test/test-version-utils/compatOptions.d.cts

```diff
@@ -21,6 +21,15 @@ export const CompatKind: {
 	DataRuntime: "DataRuntime";
 	NewDataRuntime: "NewDataRuntime";
 	LoaderDriver: "LoaderDriver";
+	/**
```

Comment:
It should also allow unification of `getCompatVersionedTestObjectProviderFromApis` and `getVersionedTestObjectProviderFromApis`. In general I think modeling it more like this will give cleaner code.

Reply:
This sounds like a good idea. I'll try to experiment with it but FYI - it may end up in a follow up PR if this one gets approved before I finish it

Reply:
I'm ok with getting the added test coverage in first and then cleaning up the API surface in a follow-up. I'd prefer that the work to clean up the API surface doesn't get backlogged though :)